.github/actions/setup-node/action.yml

```diff
@@ -0,0 +1,58 @@
+# This is the composite action:
+#   https://docs.github.com/en/actions/creating-actions/creating-a-composite-action
+#
+# Composite actions have some limitations:
+#   - many contexts are unavailable, e.g. `runner`
+#   - `env` can be specified per-step only
+#   - if `run` is used, `shell` must be explicitly specified
+name: 'Setup Node and install dependencies'
+description: 'Setup Node and install dependencies using cache'
+inputs:
+  node-version:
+    description: 'Node version'
+    required: true
+  os:
+    description: 'Composite actions do not support `runner.os`, so it must be passed in as an input'
+    required: true
+  jest-cache:
+    description: 'Whether to cache jest data'
+    required: false
+    default: false
+outputs:
+  cache-hit:
+    description: 'Cache hit for `node_modules`'
+    value: ${{ steps.node-modules-cache.outputs.cache-hit }}
+runs:
+  using: 'composite'
+  steps:
+    - name: Cache node_modules
+      id: node-modules-cache
+      uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+      with:
+        path: node_modules
+        key: node-modules-${{ inputs.os }}-${{ inputs.node-version }}-${{ hashFiles('yarn.lock') }}
+
+    - name: Setup Node
+      uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: ${{ steps.node-modules-cache.outputs.cache-hit != 'true' && 'yarn' || '' }}
```

Comment:
i don't understand this condition. you don't want to restore global yarn cache on cache hit? does it really saves time?

Reply:
I don't know the reason, but global cache is twice big compared to `node_modules` folder. Downloading and unpacking it takes some time, especially on windows workers.

<img width="859" alt="image" src="https://github.com/renovatebot/renovate/assets/1239644/a1e83dfd-f0e7-48a0-aeb1-04cf47429c42">

Reply:
Generally, this PR isn't that critical compared to unit tests run time, but this will improve "time  the first significant CI step".

Reply:
Seems like test-e2e works well without any caching, maybe let's optimize it in the following PRs